packages/astro/src/runtime/utils/random-bytes.ts

```diff
@@ -0,0 +1,58 @@
+// Based off of https://www.npmjs.com/package/@lumeweb/randombytes-browser and https://www.npmjs.com/package/randombytes
+// This rendition adds better browser support, plus better typescript support
+
+// limit of Crypto.getRandomValues()
+// https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues
+const MAX_BYTES = 65536;
+
+// Node supports requesting up to this number of bytes
+// https://github.com/nodejs/node/blob/master/lib/internal/crypto/random.js#L48
+const MAX_UINT32 = 4294967295;
+
+const GlobalCrypto = globalThis.crypto as Crypto;
+
+let randomBytes = (size: number, cb?: (...args: any) => any) => {
+    if (!(GlobalCrypto && GlobalCrypto.getRandomValues)) {
+        throw new Error(
+            'Secure random number generation is not supported by this browser.\nUse Chrome, Firefox or Internet Explorer 11'
+        );
+    }
+
+    // phantomjs needs to throw
+    if (size > MAX_UINT32) throw new RangeError('requested too many random bytes');
+
+    let bytes = new Uint32Array(size);
+
+    if (size > 0) {
+        // getRandomValues fails on IE if size == 0
+        if (size > MAX_BYTES) {
+            // this is the max bytes crypto.getRandomValues
+            // can do at once see https://developer.mozilla.org/en-US/docs/Web/API/window.crypto.getRandomValues
+            for (let generated = 0; generated < size; generated += MAX_BYTES) {
+                // buffer.slice automatically checks if the end is past the end of
+                // the buffer so we don't have to here
+                GlobalCrypto.getRandomValues(bytes.slice(generated, generated + MAX_BYTES));
+            }
+        } else {
+            GlobalCrypto.getRandomValues(bytes);
+        }
+    }
+
+    if (typeof cb === 'function') {
+        Promise.resolve().then(() => {
+            return cb(null, bytes);
+        });
+        return;
+    }
+
+    return bytes;
+};
+
+try {
+    let importStr = "crypto";
+    let cryptoImport = import(importStr);
+    randomBytes = (await cryptoImport).randomBytes as unknown as typeof randomBytes;
+} catch (e) { }
+
+export { randomBytes };
+export default randomBytes;
```

Comment:
After looking at this PR for a while, I don't think this implementation is a good idea, mainly because it uses top-level await which isn't supported in many places

Reply:
Unfortunately, the other alternative solutions aren't much better

Reply:
I have another idea, gimme a sec

Reply:
@JuanM04 Please take another look, I've removed the top-level await. I completely forgot that most of Astro's code was asynchronous in nature. I've been trying to make `serialize-javascript` with the assumption that the use case may not be asynchronous, but since most of astro's code is async in nature, my worry wasn't necessary.